app/src/main/java/org/simple/clinic/summary/bloodpressures/newbpsummary/NewBloodPressureSummaryViewEffectHandler.kt

```diff
@@ -0,0 +1,35 @@
+package org.simple.clinic.summary.bloodpressures.newbpsummary
+
+import com.spotify.mobius.rx2.RxMobius
+import io.reactivex.ObservableTransformer
+import io.reactivex.Scheduler
+import org.simple.clinic.bp.BloodPressureRepository
+import org.simple.clinic.util.scheduler.SchedulersProvider
+import javax.inject.Inject
+
+class NewBloodPressureSummaryViewEffectHandler @Inject constructor(
+    private val bloodPressureRepository: BloodPressureRepository,
+    private val schedulersProvider: SchedulersProvider
+) {
+
+  fun build(): ObservableTransformer<NewBloodPressureSummaryViewEffect, NewBloodPressureSummaryViewEvent> {
+    return RxMobius
+        .subtypeEffectHandler<NewBloodPressureSummaryViewEffect, NewBloodPressureSummaryViewEvent>()
+        .addTransformer(LoadBloodPressures::class.java, loadBloodPressureHistory(schedulersProvider.io()))
+        .build()
+  }
+
+  private fun loadBloodPressureHistory(
+      scheduler: Scheduler
+  ): ObservableTransformer<LoadBloodPressures, NewBloodPressureSummaryViewEvent> {
+    return ObservableTransformer { effect ->
+      effect
+          .switchMap {
+            bloodPressureRepository
+                .newestMeasurementsForPatient(it.patientUuid, 4)
```

Comment:
This comes back to the earlier comment I left about separating the flows for toggling the "See All" button and loading the list of BPs to display.

One problem with using magic values in codebases is that they lose meaning over time. Second is, they make testing hard since they are hardcoded and difficult to change for tests. This is an "easy" solution, but it eventually leads to making the codebase hard to maintain and understand.

I am making an educated guess here that you will display the first three BPs in this list in the UI renderer, and then check if the count is 4 or more to decide whether to toggle the "See All" button. The problem here is that the intention of displaying BPs and toggling the "See All" button are both lost to anyone who is trying to maintain the software going forward. Instead, we should attempt to make the code as explicit as possible, and make separate flows for displaying the list of BPs and toggling the "See All" button.

Reply:
I was planning on moving to a config class in later commits, but you're right I think having a config class from start will be much more meaningful. I will separate the "see all" button and loading the list of BPs effects.